pkg/operator/ceph/cr_manager.go

```diff
@@ -161,7 +161,23 @@ func (o *Operator) addToManager(m manager.Manager, c *controllerconfig.Context,
 	return nil
 }
 
+func (o *Operator) validateCRDsName() error {
+	for _, crd := range o.resources {
+		var crdNameLength int = len(crd.Name)
+		if crdNameLength > 38 {
+			return errors.New("CRD name cannot be bigger than 38 characters")
+		}
+	}
+	return nil
+}
+
 func (o *Operator) startCRDManager(context context.Context, mgrErrorCh chan error) {
+	logger.Info("check if CRDs name are valid")
```

Comment:
We need to check the CR name lengths in the admission controller. For example, in the object store CR this would be in the [ValidateObjectSpec()](https://github.com/rook/rook/blob/ef3f55460c220ec5bb902817ae063af3dad53137/pkg/apis/ceph.rook.io/v1/object.go#L72) method.

Checking at operator startup time won't prevent the CRs from being created incorrectly.

Reply:
right and actually we need to update this method https://github.com/rook/rook/blob/ef3f55460c220ec5bb902817ae063af3dad53137/pkg/apis/ceph.rook.io/v1/object.go#L63-L68
since `ValidateObjectSpec()` is called during `validate update`also which will be problem for upgrade.

Reply:
@subhamkrai why it would be a problem during upgrade

Reply:
@travisn >Checking at operator startup time won't prevent the CRs from being created incorrectly.
That right,
I thought this process will reconcile later, but if it is been created at very first then there is no need to add a cehck,
So updated the validate method 
Thanks.

Reply:
> @subhamkrai why it would be a problem during upgrade

see here https://github.com/rook/rook/pull/11233#discussion_r1008158639